contentService/src/main/java/com/kovan/app/controller/BucketController.java

```diff
@@ -0,0 +1,49 @@
+package com.kovan.app.controller;
+
+import com.kovan.app.service.BucketService;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import java.io.IOException;
+
+@RestController
+@RequestMapping("/bucket")
+public class BucketController {
+
+   private final BucketService bucketService;
+
+    public BucketController(BucketService bucketService) {
+        this.bucketService = bucketService;
+    }
+
+    @GetMapping("/download")
+    public ResponseEntity<Resource> downloadFile(@RequestParam("fileName") String fileName) throws IOException {
+        try {
+
+         Resource resource = bucketService.downloadFile(fileName);
+          String contentType = "application/octet-stream";
+
+            return ResponseEntity.ok()
+                    .contentType(MediaType.parseMediaType(contentType))
+                    .header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=\"" + resource.getFilename()+ "\"")
+                    .body(resource);
+        }
+        catch (Exception e){
+            e.printStackTrace();
```

Comment:
Don't print full stacktrace try to use any custom exception or print the error message

Reply:
printing the error message